File: OtherLang/Point.java
public class Point {
	private double x = 0;
	private double y = 0;

	public Point(double x, double y) {
		this.x = x;
		this.y = y;
	}

	public Point() {}

	public double getX() {
		return x;
	}

	public double getY() {
		return y;
	}

	public boolean equals(Point other) {
		return x == other.x && y == other.y;
	}

	public double getDistance(Point other) {
		return Math.hypot(x-other.x, y-other.y);
	}

	public String toString() {
		return "(" + x + ", " + y + ")";
	}
}

File: OtherLang/Point.php
<?php

class Point {
	private $x;
	private $y;

	public function __construct($x = 0, $y = 0) {
		$this->x = $x;
		$this->y = $y;
	}

	public function getX() {
		return $this->x;
	}

	public function getY() {
		return $this->y;
	}

	public function getDistance(Point $other) {
		$deltaX = $this->x - $other->x;
		$deltaY = $this->y - $other->y;

		return hypot($deltaX, $deltaY);
	}

	public function __toString() {
		return "($this->x, $this->y)";
	}
}
?>
